system_metrics/basic_sender.go

```diff
@@ -0,0 +1,202 @@
+		conn, err = grpc.DialContext(ctx, addr, grpc.WithInsecure())
+	} else {
+		conn, err = timber.DialCedar(ctx, &opts.Client, opts.DialOpts)
+	}
+	if err != nil {
+		return nil, errors.Wrap(err, "problem dialing rpc server")
+	}
+
+	s := &SystemMetricsClient{
+		client:     internal.NewCedarSystemMetricsClient(conn),
+		clientConn: conn,
+	}
+	return s, nil
+}
+
+// NewSystemMetricsClientWithExistingConnection returns a grpc client to send
+// system metrics data to cedar, using the provided client connection.
+func NewSystemMetricsClientWithExistingConnection(ctx context.Context, clientConn *grpc.ClientConn) (*SystemMetricsClient, error) {
+	if clientConn == nil {
+		return nil, errors.New("Must provide existing client connection")
+	}
+
+	s := &SystemMetricsClient{
+		client: internal.NewCedarSystemMetricsClient(clientConn),
+	}
+	return s, nil
+}
+
+// SystemMetricsOptions support the use and creation of a system metrics object.
+type SystemMetricsOptions struct {
+	// Unique information to identify the system metrics object.
+	Project   string `bson:"project" json:"project" yaml:"project"`
+	Version   string `bson:"version" json:"version" yaml:"version"`
+	Variant   string `bson:"variant" json:"variant" yaml:"variant"`
+	TaskName  string `bson:"task_name" json:"task_name" yaml:"task_name"`
+	TaskId    string `bson:"task_id" json:"task_id" yaml:"task_id"`
+	Execution int32  `bson:"execution" json:"execution" yaml:"execution"`
+	Mainline  bool   `bson:"mainline" json:"mainline" yaml:"mainline"`
+
+	// Data storage information for this object
+	Compression CompressionType `bson:"compression" json:"compression" yaml:"compression"`
+	Schema      SchemaType      `bson:"schema" json:"schema" yaml:"schema"`
+}
+
+// CreateSystemMetrics creates a system metrics metadata object in cedar with
+// the provided info, along with setting the created_at timestamp.
+func (s *SystemMetricsClient) CreateSystemMetricRecord(ctx context.Context, opts SystemMetricsOptions) (string, error) {
+	// validation
+	if err := opts.Compression.validate(); err != nil {
+		return "", err
+	}
+	if err := opts.Schema.validate(); err != nil {
+		return "", err
+	}
+
+	resp, err := s.client.CreateSystemMetricRecord(ctx, createSystemMetrics(opts))
+	if err != nil {
+		return "", errors.Wrap(err, "problem creating system metrics object")
+	}
+
+	return resp.Id, nil
+}
+
+// AddSystemMetricsData sends the given byte slice to the cedar backend for the
+// system metrics object with the corresponding id.
+func (s *SystemMetricsClient) AddSystemMetrics(ctx context.Context, id string, data []byte) error {
+	if id == "" {
+		return errors.New("must specify id of system metrics object")
+	}
+	if len(data) == 0 {
+		return nil
+	}
+
+	_, err := s.client.AddSystemMetrics(ctx, &internal.SystemMetricsData{
+		Id:   id,
+		Data: data,
+	})
+	return err
+}
+
+// StreamSystemMetrics is currently a no-op, will be implemented later.
```

Comment:
@julianedwards Can you make a ticket for this?

Reply:
yeah